### PR TITLE
Selects component

### PR DESCRIPTION
**Pull request by FedorLap2006**
Well, during the development of the #933 I was also busy working on Selects component implementation. Now, since all API is experimental this request will be marked as draft one. Also this PR includes some fixes and improvements: like adding WebhookMessage endpoint and files support to webhooks API ( #951). Anyways, here it is: selects component. Feel free to give a feedback on this PR.

### CODE REVIEW

**Comment by FedorLap2006** [comment]
By the way, if you see a comment and commits related to buttons, don't be confused. Since I'm waiting for Carson's review, this changes isn't merged (but this PR is based on them). So, anyways, just don't mind this comment and commits.

**Comment by fletcherist** [comment]
@FedorLap2006 Great work! Thanks, I'm gonna try to test it

**Comment by FedorLap2006** [comment]
> @FedorLap2006 Great work! Thanks, I'm gonna try to test it

Thanks. Now probably you're gonna be a little confused when you will be testing this. Because I forgot to document some moments. Probably in 30 minutes the docs will come.

**Comment by FedorLap2006** [comment]
@fletcherist So, now all docs should be in the place and you can test it if you have the beta access.

**Comment by fletcherist** [comment]
@FedorLap2006  I got this when I press button

```bash
[DG0] wsapi.go:585:onEvent() error unmarshalling INTERACTION_CREATE event, json: cannot unmarshal object into Go struct field Message.message.components of type discordgo.MessageComponent
```

**Comment by fletcherist** [comment]
I just created a message with button, and then clicked on it

**Comment by FedorLap2006** [comment]
The fixes is ready, try them out.

**Comment by fletcherist** [comment]
> The fixes is ready, try them out.

thanks for quick response. trying!

**Comment by fletcherist** [comment]
Now I've got an unmarshalling error, but event type is different
```wsapi.go:585:onEvent() error unmarshalling MESSAGE_CREATE event, json: cannot unmarshal object into Go struct field MessageCreate.components of type discordgo.MessageComponent
panic: runtime error: invalid memory address or nil pointer dereference```

**Comment by FedorLap2006** [comment]
I think API is final and we can start reviewing :)

**Comment by fletcherist** [comment]
I tested that latest commit and got error I mentioned above 
https://github.com/bwmarrin/discordgo/pull/954#issuecomment-864162952

**Comment by FedorLap2006** [comment]
Oh, yeah, forgot to push the commits.

**Comment by CarsonHoffman** [comment]
I think that the example here is getting a bit unwieldy, especially as Discord continues to add component types. It'd probably be best to have a more focused example dedicated to each type of component.

**Comment by FedorLap2006** [comment]
Huh. Sorry everyone for such long delay. Just was working on personal things. So, well, here it goes. Hope everything is done in scope of select menus, so I can move forward to other features.

**Comment by FedorLap2006** [comment]
Well, after some discussion with @CarsonHoffman we decided that I shouldn't let the branch be in the state it is and rebase it on the master branch of the repository. So, well, hope everything works. Enjoy!

File: interactions.go [hunk 1]
```diff
@@ -194,20 +194,9 @@ type InteractionData interface {
...
-	ID       string                                     `json:"id"`
-	Name     string                                     `json:"name"`
-	Resolved *ApplicationCommandInteractionDataResolved `json:"resolved"`
-	Options  []*ApplicationCommandInteractionDataOption `json:"options"`
-}
-
-// ApplicationCommandInteractionDataResolved contains resolved data for command arguments.
-// Partial Member objects are missing user, deaf and mute fields.
-// Partial Channel objects only have id, name, type and permissions fields.
-type ApplicationCommandInteractionDataResolved struct {
```
Comment (CarsonHoffman) [inline interactions.go:206]: Is there any particular reason that this PR removes this type? It still looks to be relevant from Discord's docs.

Comment (FedorLap2006) [reply interactions.go:206]: Hol' up

**Pull request merged**